tests/subsys/power/power_mgmt/src/main.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/printk.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <ztest.h>
+#include <ksched.h>
+#include <kernel.h>
+#include <power/power.h>
+#include "dummy_driver.h"
+
+#define SLEEP_MSEC 500
+#define SLEEP_TIMEOUT K_MSEC(SLEEP_MSEC)
+
+/* for checking power suspend and resume order between system and devices */
+static bool enter_low_power;
+static bool notify_app_entry;
+static bool notify_app_exit;
+static bool set_sys_power;
+static bool leave_idle;
+
+/*
+ * Weak power hook functions. Used on systems that have not implemented
+ * power management.
+ */
+__weak void sys_set_power_state(enum power_states state)
+{
+	/* set system power state, this function is called
+	 * after devices enter low power state
+	 */
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	struct device *dev;
+	uint32_t device_power_state;
+	/* at this point, devices aren't active */
+	dev = device_get_binding(DUMMY_DRIVER_NAME);
+	device_get_power_state(dev, &device_power_state);
+	zassert_false(device_power_state == DEVICE_PM_ACTIVE_STATE, NULL);
+#endif
+	zassert_true(set_sys_power == true, "App doesn't be notified");
```

Comment:
Please rephrase this message. I don't understand.

Reply:
This is still an issue. Please rephrase this message - perhaps this helps explain what is happening? "Application failed to receive notification."